tests/test_compiler.py

```diff
@@ -927,8 +927,10 @@ def ansatz():
         results0 = ref_func(*params)
         results1 = func(*params)
         if measure_f == qml.counts:
-            ndim = 2  # both [0] and m0 are on one wire only
-            results1 = {format(int(state), f"0{ndim}b"): count for state, count in zip(*results1)}
+            # Convert keys to the same data-type for comparison
+            results1 = {format(int(state), "b"): count for state, count in zip(*results1)}
+            if all(not isinstance(key, str) for key in results0.keys()):
+                results0 = {format(state, "b"): count for state, count in results0.items()}
```

Comment:
Thanks @andrijapau!

It's probably just an excess of caution, but could you please check (even locally is fine) that this change does not break any test in Catalyst?

I think of this test file as one of the closest contact points that we have with Catalyst, and it seems to me that we are transforming results data differently (please correct me if this is not the case). Thanks!

Reply:
Okay, I will try it out 😄

Reply:
Hey @PietropaoloFrisoni , I was doing some perusing in the Catalyst repository and it turns out that they have a similar test that I just stole the logic from (see [line here](https://github.com/PennyLaneAI/catalyst/blob/11e4e4f0e436fc9b051311bc10b8cef5b10f9d41/frontend/test/pytest/test_mid_circuit_measurement.py#L628). So I think it's same to merge. 🤞🏼